### PR TITLE
IGAPP-170: Show snackbar if sharing fails

### PR DESCRIPTION
**Pull request by Cmd190**
See title

### CODE REVIEW

File: native/src/components/TransparentHeader.tsx [hunk 1]
```diff
@@ -63,30 +64,29 @@ class TransparentHeader extends React.PureComponent<PropsType> {
         message
       })
     } catch (e) {
-      // TODO Show snackbar
-      reportError(e)
+      const error = e as Error
```
Comment (SteffiStuffel) [inline native/src/components/TransparentHeader.tsx:67]: Why is this needed?

**Review by SteffiStuffel — APPROVED** [review]
Tested on Android Emulator. Works as expected.

File: native/src/components/TransparentHeader.tsx [hunk 2]
```diff
@@ -63,30 +64,29 @@ class TransparentHeader extends React.PureComponent<PropsType> {
...
+      showSnackbar(error.message)
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply native/src/components/TransparentHeader.tsx:68]:
```suggestion
      showSnackbar(t('generalError')
```

**Review by Taggotty — CHANGES_REQUESTED** [review]
Please also show the snackbar for the "normal" header component

File: native/src/components/TransparentHeader.tsx (hunk 1 above)
Comment (Cmd190) [reply native/src/components/TransparentHeader.tsx:67]: If you do not cast to error explicitley you will get linting erros because  e is of type unknown and reportError() expects an Error as parameter

File: native/src/components/TransparentHeader.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Cmd190) [reply native/src/components/TransparentHeader.tsx:68]: Your're absolutley right

File: native/src/components/TransparentHeader.tsx (hunk 1 above)
Comment (klinzo) [reply native/src/components/TransparentHeader.tsx:69]: Should only be necessary starting with typescript 4.4

**Review by Taggotty — APPROVED** [review]

File: native/src/components/TransparentHeader.tsx (hunk 1 above)
Comment (Cmd190) [reply native/src/components/TransparentHeader.tsx:69]: So let's leave it for future compatability

Comment (klinzo) [reply native/src/components/TransparentHeader.tsx:69]: Actually I'd prefer to remove it again and do this all at once as we probably need this in many places and can find a general solution then.

File: native/src/components/TransparentHeader.tsx [hunk 3]
```diff
@@ -63,30 +66,27 @@ class TransparentHeader extends React.PureComponent<PropsType> {
         message
```
Comment (klinzo) [inline native/src/components/TransparentHeader.tsx:66]: Perhaps you could add a title here the same way as it is in `Header`

**Review by klinzo — COMMENTED** [review]
Could you perhaps add tests for this?

File: native/src/components/TransparentHeader.tsx (hunk 1 above)
Comment (Cmd190) [reply native/src/components/TransparentHeader.tsx:69]: ok, you've got the point

**Review by klinzo — APPROVED** [review]
